### PR TITLE
AbstractHttpClientMetricsHandler/AbstractHttpServerMetricsHandler/AccessLogHandlerH1 works now only with Buffer

### PR DESCRIPTION
**Pull request by violetagg**
Related to #1873

### CODE REVIEW

File: reactor-netty-http/src/main/java/reactor/netty/http/client/AbstractHttpClientMetricsHandler.java [hunk 1]
```diff
@@ -91,8 +91,8 @@ public Future<Void> write(ChannelHandlerContext ctx, Object msg) {
 		if (msg instanceof ByteBufHolder) {
 			dataSent += ((ByteBufHolder) msg).content().readableBytes();
 		}
-		else if (msg instanceof ByteBuf) {
-			dataSent += ((ByteBuf) msg).readableBytes();
+		else if (msg instanceof Buffer) {
+			dataSent += ((Buffer) msg).readableBytes();
```
Comment (pderop) [inline reactor-netty-http/src/main/java/reactor/netty/http/client/AbstractHttpClientMetricsHandler.java:95]: pattern matching for instanceof ?

File: reactor-netty-http/src/main/java/reactor/netty/http/client/AbstractHttpClientMetricsHandler.java [hunk 2]
```diff
@@ -113,8 +113,8 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) {
 		if (msg instanceof ByteBufHolder) {
 			dataReceived += ((ByteBufHolder) msg).content().readableBytes();
 		}
-		else if (msg instanceof ByteBuf) {
-			dataReceived += ((ByteBuf) msg).readableBytes();
+		else if (msg instanceof Buffer) {
+			dataReceived += ((Buffer) msg).readableBytes();
```
Comment (pderop) [inline reactor-netty-http/src/main/java/reactor/netty/http/client/AbstractHttpClientMetricsHandler.java:117]: same: pattern matching ?

File: reactor-netty-http/src/main/java/reactor/netty/http/server/AbstractHttpServerMetricsHandler.java [hunk 3]
```diff
@@ -97,8 +97,8 @@ public Future<Void> write(ChannelHandlerContext ctx, Object msg) {
 		if (msg instanceof ByteBufHolder) {
 			dataSent += ((ByteBufHolder) msg).content().readableBytes();
 		}
-		else if (msg instanceof ByteBuf) {
-			dataSent += ((ByteBuf) msg).readableBytes();
+		else if (msg instanceof Buffer) {
+			dataSent += ((Buffer) msg).readableBytes();
```
Comment (pderop) [inline reactor-netty-http/src/main/java/reactor/netty/http/server/AbstractHttpServerMetricsHandler.java:101]: pattern matching for instanceof ?

File: reactor-netty-http/src/main/java/reactor/netty/http/server/logging/AccessLogHandlerH1.java [hunk 4]
```diff
@@ -91,8 +91,8 @@ public Future<Void> write(ChannelHandlerContext ctx, Object msg) {
 			              }
 			          });
 		}
-		if (msg instanceof ByteBuf) {
-			accessLogArgProvider.increaseContentLength(((ByteBuf) msg).readableBytes());
+		if (msg instanceof Buffer) {
+			accessLogArgProvider.increaseContentLength(((Buffer) msg).readableBytes());
```
Comment (pderop) [inline reactor-netty-http/src/main/java/reactor/netty/http/server/logging/AccessLogHandlerH1.java:95]: pattern matching ?

**Comment by violetagg** [comment]
This will be done separately
I don't want to mix the changes

**Comment by pderop** [comment]
ah ok.

**Review by pderop — APPROVED** [review]
LGTM

**Comment by violetagg** [comment]
@pderop Thanks

**Pull request merged**